doc/specs/stdlib_terminal_color.md

```diff
@@ -0,0 +1,304 @@
+
+
+### ``fg_color_white``
+
+Foreground color enumerator representing a foreground white color escape code.
+
+
+### ``fg_color_default``
+
+Foreground color enumerator representing a foreground default color escape code.
+
+
+### ``bg_color_black``
+
+Background color enumerator representing a background black color escape code.
+
+
+### ``bg_color_red``
+
+Background color enumerator representing a background red color escape code.
+
+
+### ``bg_color_green``
+
+Background color enumerator representing a background green color escape code.
+
+
+### ``bg_color_yellow``
+
+Background color enumerator representing a background yellow color escape code.
+
+
+### ``bg_color_blue``
+
+Background color enumerator representing a background blue color escape code.
+
+
+### ``bg_color_magenta``
+
+Background color enumerator representing a background magenta color escape code.
+
+
+### ``bg_color_cyan``
+
+Background color enumerator representing a background cyan color escape code.
+
+
+### ``bg_color_white``
+
+Background color enumerator representing a background white color escape code.
+
+
+### ``bg_color_default``
+
+Background color enumerator representing a background default color escape code.
+
+
+## Procedures and methods provided
+
+### ``to_string``
+
+Generic interface to turn a style, foreground or background enumerator into an actual escape code string for printout.
+
+#### Syntax
+
+`string = [[stdlib_string_colors(module):to_string(interface)]] (code)`
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+``code``: Style, foreground or background code of ``ansi_code`` type,
+          this argument is ``intent(in)``.
+
+#### Result value
+
+The result is a default character string.
+
+#### Status
+
+Experimental
+
+#### Example
+
+```fortran
+program demo_string
+  use stdlib_terminal_colors, only : fg_color_green, style_reset, to_string
+  implicit none
+
+  print '(a)', to_string(fg_color_green) // "Colorized text message" // to_string(style_reset)
+end program demo_string
+```
+
+
+### ``operator(+)``
+
+Add two escape sequences, attributes in the right value override the left value ones.
+
+#### Syntax
+
+`code = lval + rval`
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+``lval``: Style, foreground or background code of ``ansi_code`` type,
+          this argument is ``intent(in)``.
+``rval``: Style, foreground or background code of ``ansi_code`` type,
+          this argument is ``intent(in)``.
+
+#### Result value
+
+The result is a style, foreground or background code of ``ansi_code`` type.
+
+#### Status
+
+Experimental
+
+#### Example
+
+```fortran
+program demo_combine
+  use stdlib_terminal_colors, only : fg_color_red, style_bold, ansi_code
+  implicit none
+  type(ansi_code) :: bold_red
+
+  bold_red = fg_color_red + style_bold
+end program demo_combine
+```
+
+
+### ``operator(//)``
+
+Concatenate an escape code with a string and turn it into an actual escape sequence
+
+#### Syntax
+
+`code = lval + rval`
```

Comment:
```suggestion
`colored_str = code // str`
```

Reply:
I read only later that left and right values could be either codes (in which case // is same as +) or strings. Best give a hint the return type is a string.